File: 백준/Silver/10845. 큐/큐.swift
import Foundation

let n = Int(readLine()!)!

var queue: [Int] = []
var head = 0
var output: [String] = []

let handlers: [String: ([String]) -> Void] = [
    "push": { args in
        queue.append(Int(args[0])!)
    },
    "pop": { _ in
        if head < queue.count {
            output.append(String(queue[head]))
            head += 1
        } else {
            output.append("-1")
        }
    },
    "size": { _ in
        output.append(String(queue.count - head))
    },
    "empty": { _ in
        output.append(head < queue.count ? "0" : "1")
    },
    "front": { _ in
        output.append(head < queue.count ? String(queue[head]) : "-1")
    },
    "back": { _ in
        output.append(head < queue.count ? String(queue[queue.count - 1]) : "-1")
    }
]

for _ in 0..<n {
    let parts = readLine()!.split(separator: " ").map(String.init)
    let op = parts[0]
    let args = Array(parts.dropFirst())
    handlers[op]?(args)
}

print(output.joined(separator: "\n"))

File: 백준/Silver/10845. 큐/큐.py
from collections import deque
import sys

input = sys.stdin.readline
n = int(input())
d = deque()
result = []

handlers = {
    "push": lambda args: d.append(int(args[0])),
    "pop": lambda _: result.append(str(d.popleft() if d else -1)),
    "size": lambda _: result.append(str(len(d))),
    "empty": lambda _: result.append("1" if not d else "0"),
    "front": lambda _: result.append(str(d[0] if d else -1)),
    "back": lambda _: result.append(str(d[-1] if d else -1)),
}

for _ in range(n):
    command = input().split()
    op, args = command[0], command[1:]
    handlers[op](args)

sys.stdout.write("\n".join(result))